src/main/java/com/google/sps/scheduler/Scheduler.java

```diff
@@ -0,0 +1,150 @@
+      "Review code", "Write the design doc", "Talk to PM", "Investigate report", "Prepare slides");
+
+  /**
+   * Schedules the tasks in the free time slot of the calendar events, which must be of the same day
+   * specified in the parameter.
+   * For each task that can be scheduled, a new calendar event object is created and a list of
+   * all of them is returned.
+   */
+  public static List<Event> schedule(
+      List<Event> calendarEvents, List<String> tasks, String timeZone, LocalDate dayDate) {
+    List<Event> tasksEvents = new ArrayList<>();
+
+    Set<Event> orderedCalendarEvents = new TreeSet<>(
+        Comparator.comparingLong(event -> event.getStart().getDateTime().getValue()));
+    orderedCalendarEvents.addAll(calendarEvents);
+
+    long dayStartEpochMilliseconds =
+        epochInMilliseconds(dayDate, LocalTime.of(START_HOUR, START_MINUTE), timeZone);
+    long dayEndEpochMilliseconds =
+        epochInMilliseconds(dayDate, LocalTime.of(END_HOUR, END_MINUTE), timeZone);
+
+    long lastEnd = dayStartEpochMilliseconds;
+    int indexNextTask = 0;
+
+    for (Event event : orderedCalendarEvents) {
+      long eventEnd = event.getEnd().getDateTime().getValue();
+      if (eventEnd <= dayStartEpochMilliseconds) {
+        continue;
+      }
+
+      long eventStart = event.getStart().getDateTime().getValue();
+      if (eventStart >= dayEndEpochMilliseconds) {
+        break;
+      }
+
+      while (eventStart - lastEnd >= DEFAULT_DURATION_IN_MILLISECONDS &&
+          indexNextTask < tasks.size()) {
+        DateTime startTime = epochToDateTime(lastEnd, timeZone);
+        lastEnd += DEFAULT_DURATION_IN_MILLISECONDS;
+        DateTime endTime = epochToDateTime(lastEnd, timeZone);
+        tasksEvents.add(
+            createEventWithSummary(startTime, endTime, timeZone, tasks.get(indexNextTask)));
+        indexNextTask++;
+      }
+
+      if (indexNextTask >= tasks.size()) {
+        break;
+      }
+
+      if (eventEnd > lastEnd) {
+        lastEnd = eventEnd;
+      }
+    }
+
+    while (dayEndEpochMilliseconds - lastEnd >= DEFAULT_DURATION_IN_MILLISECONDS &&
+        indexNextTask < tasks.size()) {
+      DateTime startTime = epochToDateTime(lastEnd, timeZone);
+      lastEnd += DEFAULT_DURATION_IN_MILLISECONDS;
+      DateTime endTime = epochToDateTime(lastEnd, timeZone);
+      tasksEvents.add(
+          createEventWithSummary(startTime, endTime, timeZone, tasks.get(indexNextTask)));
+      indexNextTask++;
+    }
+
+    return tasksEvents;
+  }
+
+  /**
+   * Schedules some sample tasks in the free time slot of the calendar events, which must be of
+   * the same day specified in the parameter.
+   * For each task that can be scheduled, a new calendar event object is created and a list of
+   * all of them is returned.
+   */
+  public static List<Event> schedule(List<Event> calendarEvents, String timeZone, LocalDate dayDate) {
+    return schedule(calendarEvents, TASKS_SAMPLE, timeZone, dayDate);
+  }
```

Comment:
This and the referenced fake tasks data should instead be in the test code as it is unused in production.

Reply:
They are used in the prototype to show a tasks list in the UI.

Reply:
I see. Maybe wen can then add a TODO (potentially referencing a Github issue) to the code to move this out. Alternatively you could add a class already for the Tasks integration that is coming up and let it return the fake data from there so the separation is already given.